home/import_whatsapp_templates.py

```diff
@@ -0,0 +1,205 @@
+
+        if self.purge:
+            self.delete_existing_content()
+        self.set_progress("Deleted existing WhatsApp Template", 10)
+
+        self.process_rows(rows)
+        # self.save_pages_assessment()
+
+    def process_rows(self, rows: list["ContentRow"]) -> None:
+        for i, row in reversed(list(enumerate(rows, start=2))):
+            try:
+                self.create_whatsapp_template_from_row(row)
+            except ImportWhatsAppTemplateException as e:
+                e.row_num = i
+                raise e
+
+    def create_whatsapp_template_from_row(self, row: "ContentRow") -> None:
+        locale = self.locale_from_language_code(row.locale)
+
+        template = WhatsAppTemplate(
+            name=row.name,
+            category=row.category,
+            locale=locale,
+            message=row.message,
+            example_values=[("example_values", v) for v in row.example_values],
+            submission_status=row.submission_status,
+            submission_result=row.submission_result,
+            submission_name=row.submission_name,
+        )
+
+        add_tags_to_whatsapp_template(row.quick_replies, template)
+        template.save()
+        return
+
+    def parse_file(self) -> list["ContentRow"]:
+        if self.file_type == "XLSX":
+            return self.parse_excel()
+        return self.parse_csv()
+
+    def parse_excel(self) -> list["ContentRow"]:
+        workbook = load_workbook(BytesIO(self.file_content), read_only=True)
+        worksheet = workbook.worksheets[0]
+
+        def clean_excel_cell(cell_value: str | float | datetime | None) -> str:
+            return str(cell_value).replace("_x000D", "")
+
+        first_row = next(worksheet.iter_rows(max_row=1, values_only=True))
+        header = [clean_excel_cell(cell) if cell else None for cell in first_row]
+        rows: list[ContentRow] = []
+        for row in worksheet.iter_rows(min_row=2, values_only=True):
+            r = {}
+            for name, cell in zip(header, row):  # noqa: B905 (TODO: strict?)
+                if name and cell:
+                    r[name] = clean_excel_cell(cell)
+            rows.append(ContentRow.from_flat(r))
+        # page_id_rows = group_rows_by_page_id(rows)
+        return rows
+
+    def parse_csv(self) -> list["ContentRow"]:
+        reader = csv.DictReader(StringIO(self.file_content.decode()))
+        rows = [ContentRow.from_flat(row) for row in reader]
+        # page_id_rows = group_rows_by_page_id(rows)
+        return rows
+
+    def set_progress(self, message: str, progress: int) -> None:
+        self.progress_queue.put_nowait(progress)
+
+    def delete_existing_content(self) -> None:
+        WhatsAppTemplate.objects.all().delete()
+
+    def default_locale(self) -> Locale:
+        site = Site.objects.get(is_default_site=True)
+        return site.root_page.locale
+
+
+@dataclass(slots=True, frozen=True)
+class ContentRow:
+    template_id: int | None = None
```

Comment:
We're not exporting the template ID, and we're not using it in the import either, so we can probably remove this field.